arcaflow_plugin_horreum_client/horreum_client_plugin.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+
+import sys
+import typing
+import requests
+import responses
+import json
+
+from arcaflow_plugin_sdk import plugin
+from horreum_client_schema import InputParams, SuccessOutput, ErrorOutput
+
+
+@plugin.step(
+    id="horreum-client",
+    name="Horreum Client",
+    description="Uploads an object to the Horreum server",
+    outputs={"success": SuccessOutput, "error": ErrorOutput},
+)
+def horreum_client(
+    params: InputParams,
+) -> typing.Tuple[str, typing.Union[SuccessOutput, ErrorOutput]]:
+    auth_url = (
+        params.horreum_keycloak_url + "/realms/horreum/protocol/openid-connect/token"
+    )
+    auth_obj = {
+        "username": params.horreum_username,
+        "password": params.horreum_password,
+        "grant_type": "password",
+        "client_id": "horreum-ui",
+    }
+
+    try:
+        print("==> Authenticating with keycloak...")
+        if params.mock:
+            with responses.RequestsMock() as rsps:
+                rsps.add(
+                    responses.POST,
+                    auth_url,
+                    json={"access_token": "mock_token"},
+                    status=200,
+                    content_type="application/json",
+                )
+                auth_return = requests.post(auth_url, data=auth_obj, verify=False)
+                assert auth_return.status_code == 200
```

Comment:
`assert` is generally for test code. This will generate an exception that doesn't appear to be caught. You may want to create a new ErrorOutput for this case instead of an assertion.